src/components/uploadImage/uploadImage.tsx

```diff
@@ -0,0 +1,203 @@
+import React, { useState, useEffect, useRef, useCallback } from "react";
+import styles from "./uploadImage.module.css";
+import { trpc } from "@/client/lib/trpc";
+import axios from "axios";
+import { FaCamera, FaCheckCircle } from "react-icons/fa";
+
+interface UploadImageComponentProps {
+  title: string;
+  fileName: string;
+}
+
+const UploadImageComponent: React.FC<UploadImageComponentProps> = ({
+  title,
+  fileName,
+}) => {
+  const [showStart, setShowStart] = useState<boolean>(true);
+  const [showTakePicture, setShowTakePicture] = useState<boolean>(false);
+  const [showUploadPicture, setShowUploadPicture] = useState<boolean>(false);
+  const [startButtonDisabled, setStartButtonDisabled] =
+    useState<boolean>(false);
+  const [uploading, setUploading] = useState<boolean>(false);
+  const [uploadSuccess, setUploadSuccess] = useState<boolean>(false);
+
+  const [constraints] = useState({ width: 300, height: 300 });
+  const [capturedImage, setCapturedImage] = useState<string>("");
+
+  const videoRef = useRef<HTMLVideoElement | null>(null);
+  const canvasRef = useRef<HTMLCanvasElement | null>(null);
+
+  const openUserMedia = async () => {
+    const stream = await navigator.mediaDevices.getUserMedia({ video: true });
+    const video = videoRef.current;
+    if (video !== null) {
+      video.srcObject = stream;
+      await video.play();
+    }
```

Comment:
After opening and then closing the camera component, it seems like the browser is still accessing the webcam (chrome shows the red recording circle on the tab). If possible, we should stop accessing the webcam after closing the camera. As a user, it would make me anxious to see a tab "recording" when there's nothing visually doing so on the screen.

Reply:
Hi Jonah, thanks for the feedback. I have added a stop camera function and it currently closes the camera anytime it is not in use anymore.